azure/services/aso/aso.go

```diff
@@ -53,25 +53,25 @@ type deepCopier[T any] interface {
 	DeepCopy() T
 }
 
-// Service is an implementation of the Reconciler interface. It handles creation
+// CreatorDeleter is an implementation of the Reconciler interface. It handles creation
 // and deletion of resources using ASO.
-type Service[T deepCopier[T]] struct {
+type CreatorDeleter[T deepCopier[T]] struct {
```

Comment:
I don't love this name, but wanted to change it to free up `Service` for the new abstraction.

Reply:
`ASOReconciler`? since it implements the `Reconciler` interface